src/main/java/reactor/kafka/receiver/internals/ConsumerEventLoop.java

```diff
@@ -230,12 +236,22 @@ public void run() {
                     long r = requested;
                     if (r > 0) {
                         if (!awaitingTransaction.get()) {
-                            consumer.resume(consumer.assignment());
+                            if (pausedByUs.getAndSet(false)) {
```

Comment:
can two `run()` execute concurrently? ie could it be that while we atomically switch `pausedByUs` to `false` here, the line 245 below gets executed? or is it all serialized by the `awaitingTransaction` and `requested` somehow ?

Reply:
I don't believe so, the scheduler is a `Schedulers.newSingle()`.